### ISSUE TITLE
loss nan while training retinanet_1x with iou loss

### ISSUE THREAD

**Issue by wanghao9696** [open-mmlab/mmdetection#3805]
The loss of the first epoch is normal.
When the training reaches the end of the second epoch, the loss is nan.

cfg: retinanet_r50_fpn_1x_coco.py
         loss_bbox=dict(type='IoULoss', loss_weight=1.0)

cuda10.2
pytorch1.5.1
python3.7
mmdetection v2.2.1

**Comment by v-qjqs** [comment]
Hi @wanghao9696, We have not try to use IoULoss for retinanet yet, so we do not know the reasion either. 
Could you provide a reproducible config or command for us to debug?

**Comment by wanghao9696** [comment]
My config file is set as follows：
_base_ = './retinanet_r50_fpn_1x_coco.py'
model = dict(
    bbox_head=dict(
        reg_decoded_bbox=True,
        loss_bbox=dict(type='IoULoss', loss_weight=1.0)))

My running command：
python tools/train.py  configs/aaaaa/retinanet_coco_1x/retinanet_r50_fpn_1x_coco_iou.py


In addition, training with giou_loss and diou_loss is normal. Only iou_loss have this situation. And I try two times, every time it reaches the end of the second epoch, the loss is nan.

**Comment by yuzhj** [comment]
@wanghao9696 maybe a smaller learning rate or a larger batch size can help you address this problem.
the batch size in default config is 16(2 images * 8 gpus), you need to adjust the learning rate linearly

**Comment by wanghao9696** [comment]
I'm sorry I didn't write all about configs.

I use a single GPU for training, samples_per_gpu is set to 4, and the learning rate is set to 0.0025, which is a linear setting. I trained 3 times on the coco training set. Every time when the second epoch is about to end, the loss is nan. The same configs using L1Loss, giou, and diou are all normal.

I just want to remind this situation.

**Comment by ZwwWayne** [comment]
There are indeed some cases when IoULoss is unstable. You may just use giou if it is normal.
